crates/polars-python/src/conversion/any_value.rs

```diff
@@ -399,48 +400,51 @@ pub(crate) fn py_object_to_any_value<'py>(
     ///
     /// Note: This function is only ran if the object's type is not already in the
     /// lookup table.
-    fn get_conversion_function(ob: &Bound<'_, PyAny>, py: Python<'_>) -> InitFn {
+    fn get_conversion_function(
+        ob: &Bound<'_, PyAny>,
+        py: Python<'_>,
+        allow_object: bool,
+    ) -> PyResult<InitFn> {
         if ob.is_none() {
-            get_null
+            Ok(get_null)
         }
         // bool must be checked before int because Python bool is an instance of int.
         else if ob.is_instance_of::<PyBool>() {
-            get_bool
+            Ok(get_bool)
         } else if ob.is_instance_of::<PyInt>() {
-            get_int
+            Ok(get_int)
         } else if ob.is_instance_of::<PyFloat>() {
-            get_float
+            Ok(get_float)
         } else if ob.is_instance_of::<PyString>() {
-            get_str
+            Ok(get_str)
         } else if ob.is_instance_of::<PyBytes>() {
-            get_bytes
+            Ok(get_bytes)
         } else if ob.is_instance_of::<PyList>() || ob.is_instance_of::<PyTuple>() {
-            get_list
+            Ok(get_list)
         } else if ob.is_instance_of::<PyDict>() {
-            get_struct
-        } else if ob.hasattr(intern!(py, "_s")).unwrap() {
-            get_list_from_series
```

Comment:
This was unused as far as I can tell. Removing it didn't fail any tests. The path to a literal list is elsewhere.